lib/logger.ts

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+
+function logger (...args: any): void {
+  args.unshift('[stream.new]');
+  console.log(...args); // eslint-disable-line no-console
```

Comment:
doesn't this lose line numbers?

Reply:
Line numbers in logging? Not sure what the concern is

Reply:
When console.log() fires, it records the line of the console.log() invocation rather than the line where this method was called. When walking back from sourcemaps that seems like it could be a bit opaque.

Reply:
TIL.

What would be the better way to do this?

Reply:
The best way I know of--and this is gross--is something morally like:

```
export enum LogLevel {
  DEBUG = 0,
  INFO = 1,
  WARN = 2,
  ERROR = 3,
  SILENT = 4
}

export let level: LogLevel = LogLevel.INFO;

export const debug = level >= LogLevel.DEBUG ? console.log.bind(window.console) : (params: ...any[]) => undefined;
export const info = level >= LogLevel.INFO ? console.log.bind(window.console) : (params: ...any[]) => undefined;
```

etc. etc.

It's not reliable either, and this was more of noting the potential gotcha than anything else.

Browsers are weird.

Reply:
What about the part where this prefixes `[stream.new]` to each log line.. that's mainly what I'm trying to do here

Reply:
Yeah that's not really super practical in browser JS. Which may be why people do this, I don't know. JS bad.